### PR TITLE
don't run recursive interpretation unless there is a chance of breaking

### PR DESCRIPTION
**Pull request by KristofferC**
If someone is just stepping, we might as well use compiled mode like back in the days.

Does this makes sense to you @timholy?

### CODE REVIEW

File: src/commands.jl [hunk 1]
```diff
@@ -37,10 +37,17 @@ function show_breakpoint(io::IO, bp::BreakpointRef)
     println(io)
 end
 
+const always_run_recursive_interpret = Ref(false)
+no_chance_of_breaking() = isempty(JuliaInterpreter._breakpoints) && !JuliaInterpreter.break_on_error[]
```
Comment (pfitzseb) [inline src/commands.jl]: Any reason not to use `JuliaInterpreter.breakpoints()` here?
Might also make sense to check if the breakpoints are actually enabled:
```
function no_chance_of_breaking()
  bps = JuliaInterpreter.breakpoints()
  !JuliaInterpreter.break_on_error[] && (isempty(bps) || all(bp -> !bp[].isactive, bps))
end
```

Comment (KristofferC) [reply src/commands.jl]: > Any reason not to use JuliaInterpreter.breakpoints() here?

It does a copy, which I don't really need. But of course, that copy doesnt matter and might as well use the official API.

Comment (KristofferC) [reply src/commands.jl]: >  Might also make sense to check if the breakpoints are actually enabled:

Good idea.

Comment (pfitzseb) [reply src/commands.jl]: :+1: 
Unrelated, but that's not the third or so function that lives in Juno _and_ Debugger. Maybe we should think about moving those upstream (or into DebuggerUtils.jl :P).

Comment (KristofferC) [reply src/commands.jl]: FWIW, I think some code duplication is fine.

**Comment by timholy** [comment]
Shouldn't this logic be in JuliaInterpreter? That way Juno and Rebugger get it too.

**Comment by KristofferC** [comment]
Yeah, that is very true.

**Comment by KristofferC** [comment]
Will move.

**Pull request closed without merging**